bitseq/sequence.go

```diff
@@ -273,12 +272,9 @@ func (h *Handle) set(ordinal uint32, any bool, release bool) (uint32, error) {
 			continue
 		}
 
-		// Unless unexpected error, save private copy to local copy
+		// Previous atomic push was succesfull. Save private copy to local copy
 		h.Lock()
 		defer h.Unlock()
-		if h.dbIndex != ci {
-			return ret, fmt.Errorf("unexected database index change")
-		}
```

Comment:
That might be true. But I think it is a mistake to even have a `nh` as a copy. Instead, I think you should operate just on the primary handle `h` and maybe retain the `head` if you want to revert the `h.head` back to the old head (but still dont understand why you would want to do that). And call `h.writeToStore()` instead of `nh.writeToStore()`.

But, since the current issue is a basic breakage, i would be okay to go with the current fix.
We can certainly come back and fix this properly.